src/Infrastructure/Stize.Infrastructure.Azure/Sql/SqlDatabaseExtensions.cs

```diff
@@ -67,36 +66,140 @@ public static SqlDatabaseBuilder ElasticPoolId(this SqlDatabaseBuilder builder,
+        /// <param name="databaseId">Resource ID of the primary database</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsSecondary(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        {
+            builder.Arguments.CreateMode = CreateMode.Secondary;
+            builder.Arguments.SourceDatabaseId = databaseId;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the database to be a copy of an existing database
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the database to copy</param>
         /// <returns></returns>
-        public static SqlDatabaseBuilder RestoreFrom(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        public static SqlDatabaseBuilder SetAsCopy(this SqlDatabaseBuilder builder, Input<string> databaseId)
         {
-            builder.Arguments.CreateMode = Pulumi.AzureNative.Sql.CreateMode.Restore;
+            builder.Arguments.CreateMode = CreateMode.Copy;
             builder.Arguments.SourceDatabaseId = databaseId;
             return builder;
         }
 
         /// <summary>
-        /// Enables the read scale
+        /// Sets the database to be a restoration of a geo-replicated backup database.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Recoverable Database Resource ID of the database to restore</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsRecovery(this SqlDatabaseBuilder builder, Input<string> databaseId)
+        {
+            /* TODO: May need improvement and needs to be properly tested!
+             * User must pass in the recoverableDatabaseId, which is associated with a database stored in the recoverable databases of the server.
+             * Format of recoverableDatabaseId:
+             * 'subscriptions/subscription-id/resourceGroups/resource-group-name/providers/Microsoft.Sql/servers/server-name/recoverableDatabases/recoverable-database-name'
+            */
+            builder.Arguments.CreateMode = CreateMode.Recovery;
+            builder.Arguments.RecoverableDatabaseId = databaseId;
+            return builder;
+        }
+
+        /// <summary>
+        /// Sets the database to be a restoration of a point in time of an existing database, using the existing database's resource ID and the point in time (ISO8601 format) to restore from.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="databaseId">Resource ID of the database to restore from</param>
+        /// <param name="restorePointInTime">The point in time (ISO8601 format) of the source database</param>
+        /// <returns></returns>
+        public static SqlDatabaseBuilder SetAsPointInTimeRestore(this SqlDatabaseBuilder builder, Input<string> databaseId, Input<string> restorePointInTime)
```

Comment:
Should we rename this to "CreateAsPointInTimeRestore"? I feel the current name is a bit confusing.